test/DeploymentTemplate.test.ts

```diff
@@ -949,3 +950,101 @@ suite("ReferenceInVariableDefinitionJSONVisitor", () => {
         });
     });
 });
+
+suite("Incomplete JSON shouldn't crash parse", () => {
+    const template: string =
+        `{
+        "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+        "contentVersion": "1.0.0.0",
+        "parameters": {
+            "location": { "type": "string" },
+            "networkInterfaceName": {
+                "type": "string"
+            },
+        },
+        "variables": {
+            "vnetId": "[resourceId(resourceGroup().name,'Microsoft.Network/virtualNetworks', parameters('virtualNetworkName'))]",
+        },
+        "resources": [
+            {
+                "name": "[parameters('networkInterfaceName')]",
+                "type": "Microsoft.Network/networkInterfaces",
+                "apiVersion": "2018-10-01",
+                "location": "[parameters('location')]",
+                "dependsOn": [
+                    "[concat('Microsoft.Network/networkSecurityGroups/', parameters('networkSecurityGroupName'))]",
+                    "[concat('Microsoft.Network/virtualNetworks/', parameters('virtualNetworkName'))]",
+                    "[concat('Microsoft.Network/publicIpAddresses/', parameters('publicIpAddressName'))]"
+                ],
+                "properties": {
+                    "$test-commandToExecute": "[concat('cd /hub*/docker-compose; sudo docker-compose down -t 60; sudo -s source /set_hub_url.sh ', reference(parameters('publicIpName')).dnsSettings.fqdn, ';  sudo docker volume rm ''dockercompose_cert-volume''; sudo docker-compose up')]",
+                    "ipConfigurations": [
+                        {
+                            "name": "ipconfig1",
+                            "properties": {
+                                "subnet": {
+                                    "id": "[variables('subnetRef')]"
+                                },
+                                "privateIPAllocationMethod": "Dynamic",
+                                "publicIpAddress": {
+                                    "id": "[resourceId(resourceGroup().name, 'Microsoft.Network/publicIpAddresses', parameters('publicIpAddressName'))]"
+                                }
+                            }
+                        }
+                    ]
+                },
+                "tags": {}
+            }
+        ],
+        "outputs": {
+            "adminUsername": {
+                "type": "string",
+                "value": "[parameters('adminUsername')]"
+            }
+        }
+    }
+    `;
+
+    test("https://github.com/Microsoft/vscode-azurearmtools/issues/193", () => {
+        let modifiedTemplate = template.replace('"type": "string"', '"type": string');
+        new DeploymentTemplate(modifiedTemplate, "id");
```

Comment:
Any particular reason the other tests do `await dt.errors;` but this one doesn't?

Reply:
Not really, the bug occurs during new DeploymentTemplate, I'll add it here, though.